src/content/docs/en/guides/typescript.mdx

```diff
@@ -83,15 +84,35 @@ import type { SomeType } from './script';
 
 This way, you avoid edge cases where Astro's bundler may try to incorrectly bundle your imported types as if they were JavaScript.
 
-In your `.tsconfig` file, you can instruct TypeScript to help with this. The [`verbatimModuleSyntax` setting](https://www.typescriptlang.org/tsconfig#verbatimModuleSyntax) can be set to `true`. Then, TypeScript will check your imports and tell you when  `import type` should be used. This setting is included by default in our `strict` and `strictest` templates.
+In your `.tsconfig` file, you can instruct TypeScript to help with this.
 
-```json title="tsconfig.json" ins={3}
-{
-  "compilerOptions": {
-    "verbatimModuleSyntax": true,
-  }
-}
-```
+<Tabs client:visible sharedStore="typescript-version">
+	<Fragment slot="tab.4.9">TypeScript ≤4.9</Fragment>
+	<Fragment slot="tab.5.0">TypeScript ≥5.0</Fragment>
+
+	<Fragment slot="panel.4.9">
+		Set [`importsNotUsedAsValues`](https://www.typescriptlang.org/tsconfig#importsNotUsedAsValues) to `"error"`. TypeScript will check your imports and tell you when `import type` should be used. This setting is included by default in our `strict` and `strictest` templates.
+
+    ```json title="tsconfig.json" ins={3}
+    {
+      "compilerOptions": {
+        "importsNotUsedAsValues": "error",
+      }
+    }
+    ```
+	</Fragment>
+	<Fragment slot="panel.5.0">
+		Set [`verbatimModuleSyntax`](https://www.typescriptlang.org/tsconfig#verbatimModuleSyntax) to `true`. TypeScript will check your imports and tell you when `import type` should be used. This setting is included by default in our `strict` and `strictest` templates.
```

Comment:
It is not true that this setting is included in our template, `importsNotUsedAsValues` is with another setting preventing it from erroring on 5.0.

Not sure if it really matters since they kinda do the same thing (though `verbatimModuleSyntax` is just better), but still, technicalities